README.md

```diff
@@ -2,31 +2,120 @@
+out, that would be bad news. When creating applications that handle passwords
+and storing these passwords, we need to be careful with how this information is
+passed around.
 
-Discuss the topic of the reading. Break into sections and sub-sections as
-appropriate to help with the reader's orientation with the material and future
-referencing.
+So far, we have been storing our passwords in plain text. Look at our database
+where we stored the passwords for our `mary` and `admin` users in plain, visible
+text.
 
-## Topic 2
+How can we protect our passwords and other sensitive data in a more secure way?
+Let's first define some common words when it comes to securing passwords.
 
-Discuss the topic of the reading. Break into sections and sub-sections as
-appropriate to help with the reader's orientation with the material and future
-referencing.
+## Encoding, Encrypting, and Hashing
+
+Encoding and encrypting and hashing - oh my!
+
+These security buzzwords are usually confused to mean the same thing. But these
+actually are different techniques. Let's look at the definitions of each:
+
+**Encoding**: This technique transforms data from one format to another so that
+it can be consumed by other systems. This technique is not used for security
+purposes and can be easily reversed. An example of encoding is if we look at an
+ASCII table:
+
+![ASCII Table](https://curriculum-content.s3.amazonaws.com/java-mod-2/chars-and-strings/ASCII-Table.png)
+
+[Medium: ASCII Table in Java Programming](https://medium.com/@aidafarihabaharunsuratman/did-someone-actually-use-ascii-table-in-java-programming-9710a65c6ed9)
+
+We can easily translate the uppercase character `A` to the decimal value of 65
+and back.
+
+**Encrypting**: This technique makes the data unreadable and difficult to decode
+to an unauthorized reader. It makes use of a secret key in order to transform
+the data back to its original state.
+
+A simple example of encryption: Say Suzie wants to send a secret message to
+Dustin. She doesn't want to encode it in case something evil tries to intercept
+it. In order to do so, she creates a key to go along with the message. Dustin
+also knows the key, so he can decrypt the message to read it. If it was
+intercepted, the message would look like gibberish to the unauthorized
+recipient as long as they also don't have the key.
+
+So while encryption is used for security purposes, it can also be reversed if
+the key is known.
+
+**Hashing**: This technique takes the data and generates a `String` that is
+specifically based on the input of the data. If the input is changed, even just
+slightly, the hash will also change. This ensures data integrity. We can
+generate this hash by using a hashing algorithm. Once the input is put through
+the hashing algorithm, the data _cannot_ be reversed back. This makes this
+technique one of the most secure ways to handle passwords!
+
+## Password Hashing in PostgreSQL
```

Comment:
Would it be easier for the students to look at the output of the Java BCryptPasswordEncoder bean's encode method instead of using a Postgres extension?